test/integration/compiler.spec.ts

```diff
@@ -0,0 +1,36 @@
+import { Expect, Test, TestCase } from "alsatian";
+import { execSync } from "child_process"
+import { existsSync, removeSync, unlink } from "fs-extra"
+import { resolve } from "path"
+
+export class CompilerTests {
+
+    @TestCase('tsconfig.default.json', ['typescript_lualib.lua', 'nested/folder/file.lua'])
+    @TestCase('tsconfig.outDir.json', ['outDir/typescript_lualib.lua', 'outDir/nested/folder/file.lua'])
+    @TestCase('tsconfig.rootDir.json', ['nested/typescript_lualib.lua', 'nested/folder/file.lua'])
+    @TestCase('tsconfig.bothDirOptions.json', ['outDir/typescript_lualib.lua', 'outDir/folder/file.lua'])
+    @Test("Compile project")
+    public compileProject(tsconfig: string, expectedFiles: string[]) {
+        const compilerPath = resolve('dist/Compiler.js');
+        const tsconfigPath = resolve('test/integration/project', tsconfig);
+
+        // Compile project
+        execSync(`node ${compilerPath} -p ${tsconfigPath}`);
```

Comment:
Don't use exec to test. Errors won't be reported correctly also coverage reports probably won't work either.

Reply:
What should I use instead?

Reply:
function executeCommandLine(args: ReadonlyArray<string>)

Reply:
I can't figure this out, I tried passing a callback but it never gets called.

Reply:
to be fixed later #68